apiserver/facades/client/sshclient/facade.go

```diff
@@ -289,23 +294,32 @@ func (facade *Facade) getMachineForEntity(ctx context.Context, tagString string)
 
 	switch tag := tag.(type) {
 	case names.MachineTag:
-		m, err := facade.backend.Machine(tag.Id())
+		machineName := machine.Name(tag.Id())
+		machineUUID, err := facade.machineService.GetMachineUUID(ctx, machineName)
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
-		return &sshMachine{Machine: m, networkService: facade.networkService}, nil
+		return &sshMachine{
+			machineUUID:    machineUUID,
+			machineName:    machineName,
+			networkService: facade.networkService,
+		}, nil
 	case names.UnitTag:
 		machineName, err := facade.applicationService.GetUnitMachineName(ctx, unit.Name(tag.Id()))
 		if errors.Is(err, applicationerrors.UnitNotFound) {
 			return nil, errors.NotFoundf("unit %q", tag.Id())
 		} else if err != nil {
 			return nil, errors.Trace(err)
 		}
-		m, err := facade.backend.Machine(machineName.String())
+		machineUUID, err := facade.machineService.GetMachineUUID(ctx, machineName)
```

Comment:
Both cases end up getting the machineUUID, you could remove this switch altogether, right?